### PR TITLE
Make variadic bindings optional + fixes

### PR DESCRIPTION
**Pull request by ghost**
Add missing arity checks in bindings evaluator
Update arity error messages for bindings/assignments
Remove redundant arityMismatchError function
Update tests

### CODE REVIEW

File: test/runtests.py [hunk 1]
```diff
@@ -389,11 +389,13 @@ def runTests(opt):
...
         print "[Tests]"
         for test in testcases:
+            startTime = time.time()
             res, log = results.next()
+            endTime = time.time()
             if res != "ok" and res != "disabled":
                 logfile.write(log)
             logfile.flush()
-            print "%s: %s" % (test.name(), res)
+            print "%s in %fs: %s" % (test.name(), (endTime - startTime), res)
```
Comment (jckarter) [inline test/runtests.py]: Can you move the timing after the `:` so the output is still parseable by python's `config` module? Something like `<name>: <status> (<time>)` perhaps

File: test/multivalues/assignmismatch/2/compilererr.txt [hunk 2]
```diff
@@ -1 +1 @@
-left side has 1 value, but right side has 2 values
+error: expected 1 value, but received 2 values
```
Comment (jckarter) [inline test/multivalues/assignmismatch/2/compilererr.txt]: This seems like a step backward in error message quality.

File: test/runtests.py (hunk 1 above)
Comment (ghost) [inline test/runtests.py]: Oh, meant to delete the timing stuff. Is it worth leaving in?

Comment (jckarter) [inline test/runtests.py]: Seems useful to me.

**Pull request merged**